packages/plugin-typeorm/src/typeorm.ts

```diff
@@ -1,31 +1,60 @@
-import { BasePlugin } from '@opentelemetry/core';
-import { Span, SpanKind, StatusCode, setSpan, context } from '@opentelemetry/api';
+import { Span, SpanKind, StatusCode, setSpan, context, Logger } from '@opentelemetry/api';
 import { DatabaseAttribute, GeneralAttribute } from '@opentelemetry/semantic-conventions';
 import { TypeormPluginConfig } from './types';
-import { safeExecute, getParamNames } from './utils';
-import shimmer from 'shimmer';
+import { getParamNames } from './utils';
 import { VERSION } from './version';
-import * as typeorm from 'typeorm';
+import type * as typeorm from 'typeorm';
+import {
+    InstrumentationBase,
+    InstrumentationConfig,
+    InstrumentationModuleDefinition,
+    InstrumentationNodeModuleDefinition,
+    isWrapped,
+    safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+
+type Config = InstrumentationConfig & TypeormPluginConfig;
+
+export class TypeormInstrumentation extends InstrumentationBase<typeof typeorm> {
+    static readonly component = 'typeorm';
+    protected _config!: Config;
+
+    constructor(config: Config = {}) {
+        super('opentelemetry-instrumentation-typeorm', VERSION, Object.assign({}, config));
+    }
 
-class TypeormPlugin extends BasePlugin<typeof typeorm> {
-    protected _config!: TypeormPluginConfig;
+    setConfig(config: Config = {}) {
+        this._config = Object.assign({}, config);
+    }
 
-    constructor(readonly moduleName: string) {
-        super(`opentelemetry-plugin-typeorm`, VERSION);
+    get logger() {
+        return this._config.logger ?? this._logger ?? ({} as Logger);
```

Comment:
what will happen if we fall to the last category `{} as Logger` and then call something like `error` on it?